backend/src/v5/processors/teamspaces/projects/models/commons/tickets.groups.js

```diff
@@ -79,22 +105,126 @@ const getObjectArrayFromRules = async (teamspace, project, model, revId, rules)
+		const projection = { metadata: { $elemMatch: { $or: externalIdKeys.map((n) => ({ key: n })) } } };
+		const metadata = await getMetadataByQuery(teamspace, obj.container, query, projection);
+
+		if (metadata?.length) {
+			delete convertedObject._ids;
+			const externalIdName = getExteralidNameFromMetadata(metadata);
+			convertedObject[externalIdName] = [...new Set(metadata.map((m) => m.metadata[0].value))];
+		}
+
+		return convertedObject;
+	}));
+
+	return convertedObjects;
+};
+
+const convertExternalIdsTo3dRepoIds = async (teamspace, project, revId, objects) => {
+	const convertedObjects = await Promise.all(objects.map(async (obj) => {
+		if (obj._ids) {
+			return obj;
+		}
+
+		const convertedObject = { ...obj };
+		const { container } = obj;
+
+		let revision = revId;
+		if (!revision) {
+			try {
+				const rev = await getLatestRevision(teamspace, container, { _id: 1 });
+				revision = rev._id;
+			} catch (err) {
+				return [];
+			}
+		}
+
+		const externalIdName = obj.revit_ids ? 'revit_ids' : 'ifc_guids';
+		const query = {
+			rev_id: revision,
+			metadata: { $elemMatch: { key: { $in: externalIdNamesToKeys[externalIdName] },
+				value: { $in: obj[externalIdName] } } },
+		};
+
+		const metadata = await getMetadataByQuery(teamspace, container, query, { parents: 1 });
+		const nodes = await getNodesBySharedIds(teamspace, project, container, revision,
+			metadata.flatMap(({ parents }) => parents), { _id: 1 });
+
+		delete convertedObject[externalIdName];
+		convertedObject._ids = [];
+
+		const idToMeshes = await getIdToMeshesMapping(teamspace, container, revision);
+		nodes.forEach(({ _id }) => {
+			const idStr = UUIDToString(_id);
+			if (idToMeshes[idStr]) {
+				idToMeshes[idStr].forEach((id) => {
+					convertedObject._ids.push(stringToUUID(id));
+				});
+			}
+		});
+
+		return convertedObject;
+	}));
+
+	return convertedObjects.flat();
+};
+
+TicketGroups.addGroups = async (teamspace, project, model, ticket, groups) => {
+	const convertedGroups = await Promise.all(groups.map(
+		async (group) => {
+			if (group.objects) {
+				const objects = await convert3dRepoIdsToExternalIds(teamspace, project, group.objects);
+				return { ...group, objects };
+			}
+
+			return group;
+		}));
+
+	await addGroups(teamspace, project, model, ticket, convertedGroups);
+};
+
+TicketGroups.updateTicketGroup = async (teamspace, project, model, ticket, groupId, data, author) => {
+	const convertedData = { ...data };
 
-TicketGroups.getTicketGroupById = async (teamspace, project, model, revId, ticket, groupId, containers) => {
+	if (data.objects) {
+		convertedData.objects = await convert3dRepoIdsToExternalIds(teamspace, project, data.objects);
+	}
+
+	await updateGroup(teamspace, project, model, ticket, groupId, convertedData, author);
+};
+
+TicketGroups.getTicketGroupById = async (teamspace, project, model, revId, ticket, groupId, convertTo3dRepoIds,
```

Comment:
I swapped the order between `convertTo3DRepoGuids` and containers, this broke federations previously (discovered whilst writing e2e tests).

I probably broke unit tests with the change, please fix